internal/test/framework/framework.go

```diff
@@ -294,6 +301,73 @@ func (f *TestFramework) GetNode(externalIP string) (*v1.Node, error) {
 	return matchedNode, nil
 }
 
+// GetNode uses external IP and finds out the name associated with the node
+func (f *TestFramework) GetNodeName(externalIP string) string {
+	node, err := f.GetNode(externalIP)
+	if err != nil {
+		log.Printf("error while getting required kubernetes node object: %v", err)
+		return ""
+	}
+	return node.Name
+}
+
+// WriteToArtifactDir will write contents to $ARTIFACT_DIR/subDirName/filename. If subDirName is empty, contents
+// will be written to $ARTIFACT_DIR/filename
+func (f *TestFramework) WriteToArtifactDir(contents []byte, subDirName, filename string) error {
+	path := filepath.Join(artifactDir, subDirName, filename)
+	dir, _ := filepath.Split(path)
+	err := os.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("could not create %s: %s", dir, err)
+	}
+	return ioutil.WriteFile(path, contents, os.ModePerm)
+}
+
+// RetrieveArtifacts should retrieve artifacts related the test run. Ideally this should retrieve all the logs related
+// to the Windows VM. This shouldn't return an error but should print the failures as log collection is nice to have
+// rather than a must have.
+// TODO: Think about how we can retrieve stdout from ansible out within this function
+func (f *TestFramework) RetrieveArtifacts() {
+	for _, vm := range f.WinVMs {
+		if vm == nil {
+			continue
+		}
+		if vm.GetCredentials() == nil {
+			log.Print("error while getting credentials of the vm")
+			continue
+		}
+
+		instanceID := vm.GetCredentials().GetInstanceId()
+		if len(instanceID) == 0 {
+			log.Print("error while getting instance id of the vm")
+			continue
+		}
+
+		externalIP := vm.GetCredentials().GetIPAddress()
+		if len(externalIP) == 0 {
+			log.Printf("error while getting external ip address of the vm with instance ID %s", instanceID)
+			continue
+		}
+
+		nodeName := f.GetNodeName(externalIP)
+		if len(nodeName) == 0 {
+			log.Printf("error while getting internal ip address of the node object: %s", instanceID)
+		}
+
+		// We want a format like "ip-10-0-141-99.ec2.internal/nodes/logs/wsu/kubelet"
```

Comment:
nodes/ip-10-0-141-99.ec2.internal/logs...

Reply:
Thanks for catching it.